base/tuple.jl

```diff
@@ -569,9 +569,14 @@ end
 # for comparison. This is useful for sorting tuples of integers.
 # TODO: remove this when the compiler can optimize the generic version better
 # See #48724 and #48753
-_pack_tuple((a,b)) = widen((a)) << 8sizeof(b) - typemin(b) + b
-for T in unique((BitIntegerSmall_types..., Int, UInt))
-    @eval isless(a::NTuple{2, $T}, b::NTuple{2, $T}) = isless(_pack_tuple(a), _pack_tuple(b))
+_pack_tuple_promote(a::Unsigned,b) = unsigned(promote(a,b)[1])
+_pack_tuple_promote(a::Signed,b) = signed(promote(a,b)[1])
```

Comment:
I could not get the adaptive widening to work using promotion, thats why I used the @generated function.
This can result in calling `promote(UInt8(200),Int8(-1))`, which errors with `InexactError: check_top_bit(UInt8, -1)`.